### PR TITLE
Limit number of parallel jar transforms to available processors

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public DefaultCachedClasspathTransformer(
         this.globalCacheLocations = globalCacheLocations;
         this.cache = classpathTransformerCacheFactory.createCache(cacheRepository, fileAccessTimeJournal);
         this.fileAccessTracker = classpathTransformerCacheFactory.createFileAccessTracker(fileAccessTimeJournal);
-        this.executor = executorFactory.create("jar transforms");
+        this.executor = executorFactory.create("jar transforms", Runtime.getRuntime().availableProcessors());
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java:74]: Shouldn't we be using worker leases here?

Comment (bamboo) [reply subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java:74]: I don't know. Can you make the case?

Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java:74]: My understanding is that we want all parallel work to be governed by worker leases. That way we can ensure that there's always at most `maxWorkers` number of parallel things happening at the same time. This code here seems no exception to that.

Comment (bamboo) [reply subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java:74]: @lptr, that's a good case. I've added #16181 so we can revisit it after we're done with all things 7.0.

**Pull request merged**